mobile/photos.go

```diff
@@ -164,17 +164,30 @@ func (m *Mobile) GetPhotoData(id string, path string) (string, error) {
 		log.Errorf("get block data failed %s: %s", id, err)
 		return "", err
 	}
-	_, formatStr, err := image.DecodeConfig(bytes.NewReader(data))
+	var format string
+	meta, err := thrd.GetPhotoMetaData(id, block)
 	if err != nil {
-		log.Errorf("could not determine image format: %s", err)
-		return "", err
+		log.Warningf("get indexed photo meta data failed, decoding...")
+		_, format, err = image.DecodeConfig(bytes.NewReader(data))
+		if err != nil {
+			log.Errorf("could not determine image format: %s", err)
+			return "", err
+		}
+	} else {
+		format = meta.EncodingFormat
 	}
-	prefix := getImageDataURLPrefix(util.Format(formatStr))
+	prefix := getImageDataURLPrefix(util.Format(format))
 	encoded := libp2pc.ConfigEncodeKey(data)
 	img := &ImageData{Url: prefix + encoded}
 	return toJSON(img)
 }
 
+// GetPhotoDataForSize returns a data url of an image at or above requested size, or the next best option
+func (m *Mobile) GetPhotoDataForMinWidth(id string, minWidth int) (string, error) {
```

Comment:
Awesome. Thinking through the consequences of only passing width instead of width and height... Close enough with just width?

Reply:
That's what I was thinking... but maybe useful? Whatever you guys think. The other option for restricting via height I guess would be css + auto margin for centering